### PR TITLE
Implement `mkdirpPromise`

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
- [x] code is properly formatted (`npm run fmt`)
- [x] tests are added/updated
- [x] documentation is updated (`npm run doc` to regenerate documentation based on comments)
- [x] description of changes is added under the `Unreleased` header in CHANGELOG.md

### CODE REVIEW

File: lib/fs.js [hunk 1]
```diff
@@ -129,8 +132,26 @@ const rmRecursivePromise = async path =>
     }
   );
 
+const mkdirPromise = async (dir, mode) =>
+  fsPromises.access(dir).then(
+    () => Promise.resolve(),
+    async () => {
+      await mkdirPromise(pathModule.dirname(dir, mode));
```
Comment (belochub) [inline lib/fs.js]: This doesn't quite make sense, `path.dirname()` only accepts a single argument.

**Comment by SemenchenkoVitaliy** [comment]
ping @belochub

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@ and this project adheres to
 
 ## [Unreleased][unreleased]
 
+### Added
+
+- `mkdirpPromise` function.
```
Comment (belochub) [inline CHANGELOG.md]:
```suggestion
- `common.mkdirpPromise()` function.
```

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/common/commit/56eddc0c244f5e093e1cd24ca8c11c410a2675d9.

**Pull request closed without merging**